### PR TITLE
Docs: add possible value list of property for 43 resource

### PR DESCRIPTION
**Pull request by wuxu92**
Some Property has validator check in the schema definition, but are missing in the document, this PR try to fix all these kind of issue.

### CODE REVIEW

File: website/docs/r/frontdoor.html.markdown [hunk 1]
```diff
@@ -167,7 +167,7 @@ The `backend_pool_health_probe` block supports the following:
 
 * `path` - (Optional) The path to use for the Health Probe. Default is `/`.
 
-* `protocol` - (Optional) Protocol scheme to use for the Health Probe. Defaults to `Http`.
+* `protocol` - (Optional) Protocol scheme to use for the Health Probe. Defaults to `Http`. Possible values are `Http` and `Https`.
```
Comment (stephybun) [inline website/docs/r/frontdoor.html.markdown:170]: Can we switch the order around for the possible values and the default value

```suggestion
* `protocol` - (Optional) Protocol scheme to use for the Health Probe. Possible values are `Http` and `Https`. Defaults to `Http`.
```

File: website/docs/r/batch_pool.html.markdown [hunk 2]
```diff
@@ -160,7 +160,7 @@ The following arguments are supported:
...
+* `os_disk_placement` - (Optional) Specifies the ephemeral disk placement for operating system disk for all VMs in the pool. This property can be used by user in the request to choose which location the operating system should be in. e.g., cache disk space for Ephemeral OS disk provisioning. For more information on Ephemeral OS disk size requirements, please refer to Ephemeral OS disk size requirements for Windows VMs at <https://docs.microsoft.com/en-us/azure/virtual-machines/windows/ephemeral-os-disks#size-requirements> and Linux VMs at <https://docs.microsoft.com/en-us/azure/virtual-machines/linux/ephemeral-os-disks#size-requirements>. Possible value is `CacheDisk`.
```
Comment (stephybun) [inline website/docs/r/batch_pool.html.markdown:163]: Can we change this to 
```suggestion
* `os_disk_placement` - (Optional) Specifies the ephemeral disk placement for operating system disk for all VMs in the pool. This property can be used by the user in the request to choose which location the operating system should be in. e.g. cache disk space for Ephemeral OS disk provisioning. For more information on Ephemeral OS disk size requirements, please refer to Ephemeral OS disk size requirements for Windows VMs at <https://docs.microsoft.com/en-us/azure/virtual-machines/windows/ephemeral-os-disks#size-requirements> and Linux VMs at <https://docs.microsoft.com/en-us/azure/virtual-machines/linux/ephemeral-os-disks#size-requirements>. The only possible value is `CacheDisk`.
```

**Review by stephybun — CHANGES_REQUESTED** [review]
Thanks for this @wuxu92

I left two comments regarding cases where a property only accepts one value, and also regarding the ordering of information in the property description when there are possible values and a default value. Could you please apply those suggestions to all the properties in this PR?

File: website/docs/r/frontdoor.html.markdown (hunk 1 above)
Comment (wuxu92) [reply website/docs/r/frontdoor.html.markdown]: updated

File: website/docs/r/virtual_network_gateway_connection.html.markdown [hunk 3]
```diff
@@ -247,11 +247,9 @@ The following arguments are supported:
 * `shared_key` - (Optional) The shared IPSec key. A key could be provided if a
     Site-to-Site, VNet-to-VNet or ExpressRoute connection is created.
 
-* `connection_mode` - (Optional) Connection mode to use. Possible
-values are `Default`, `InitiatorOnly` and `ResponderOnly`. Defaults to `Default`.
-Changing this value will force a resource to be created.
+* `connection_mode` - (Optional) Connection mode to use. Possible values are `Default`, `InitiatorOnly` and `ResponderOnly`. Defaults to `Default`. Changing this value will force a resource to be created.
 
-* `connection_protocol` - (Optional) The IKE protocol version to use. Possible
+* `connection_protocol` - (Optional) The IKE protocol version to use. PossiblePossible values are `IKEv1` and `IKEv2`.
```
Comment (stephybun) [inline website/docs/r/virtual_network_gateway_connection.html.markdown:252]:
```suggestion
* `connection_protocol` - (Optional) The IKE protocol version to use. Possible values are `IKEv1` and `IKEv2`.
```

File: website/docs/r/web_application_firewall_policy.html.markdown [hunk 4]
```diff
@@ -132,11 +132,11 @@ The `custom_rules` block supports the following:
 
 * `priority` - (Required) Describes priority of the rule. Rules with a lower value will be evaluated before rules with a higher value.
 
-* `rule_type` - (Required) Describes the type of rule.
+* `rule_type` - (Required) Describes the type of rule.Possible values are `MatchRule` and `Invalid`.
```
Comment (stephybun) [inline website/docs/r/web_application_firewall_policy.html.markdown:135]:
```suggestion
* `rule_type` - (Required) Describes the type of rule. Possible values are `MatchRule` and `Invalid`.
```

File: website/docs/r/web_application_firewall_policy.html.markdown [hunk 5]
```diff
@@ -132,11 +132,11 @@ The `custom_rules` block supports the following:
...
 
 * `match_conditions` - (Required) One or more `match_conditions` blocks as defined below.
 
-* `action` - (Required) Type of action.
+* `action` - (Required) Type of action.Possible values are `Allow`, `Block` and `Log`.
```
Comment (stephybun) [inline website/docs/r/web_application_firewall_policy.html.markdown:139]:
```suggestion
* `action` - (Required) Type of action. Possible values are `Allow`, `Block` and `Log`.
```

File: website/docs/r/web_application_firewall_policy.html.markdown [hunk 6]
```diff
@@ -146,17 +146,17 @@ The `match_conditions` block supports the following:
 
 * `match_values` - (Required) A list of match values.
 
-* `operator` - (Required) Describes operator to be matched.
+* `operator` - (Required) Describes operator to be matched.Possible values are `IPMatch`, `GeoMatch`, `Equal`, `Contains`, `LessThan`, `GreaterThan`, `LessThanOrEqual`, `GreaterThanOrEqual`, `BeginsWith`, `EndsWith` and `Regex`.
```
Comment (stephybun) [inline website/docs/r/web_application_firewall_policy.html.markdown:149]:
```suggestion
* `operator` - (Required) Describes operator to be matched. Possible values are `IPMatch`, `GeoMatch`, `Equal`, `Contains`, `LessThan`, `GreaterThan`, `LessThanOrEqual`, `GreaterThanOrEqual`, `BeginsWith`, `EndsWith` and `Regex`.
```

File: website/docs/r/web_application_firewall_policy.html.markdown [hunk 7]
```diff
@@ -146,17 +146,17 @@ The `match_conditions` block supports the following:
...
 
 * `negation_condition` - (Optional) Describes if this is negate condition or not
 
-* `transforms` - (Optional) A list of transformations to do before the match is attempted.
+* `transforms` - (Optional) A list of transformations to do before the match is attempted.Possible values are `HtmlEntityDecode`, `Lowercase`, `RemoveNulls`, `Trim`, `UrlDecode` and `UrlEncode`.
```
Comment (stephybun) [inline website/docs/r/web_application_firewall_policy.html.markdown:153]:
```suggestion
* `transforms` - (Optional) A list of transformations to do before the match is attempted. Possible values are `HtmlEntityDecode`, `Lowercase`, `RemoveNulls`, `Trim`, `UrlDecode` and `UrlEncode`.
```

File: website/docs/r/web_application_firewall_policy.html.markdown [hunk 8]
```diff
@@ -146,17 +146,17 @@ The `match_conditions` block supports the following:
...
 
 ---
 
 The `match_variables` block supports the following:
 
-* `variable_name` - (Required) The name of the Match Variable
+* `variable_name` - (Required) The name of the Match VariablePossible values are `RemoteAddr`, `RequestMethod`, `QueryString`, `PostArgs`, `RequestUri`, `RequestHeaders`, `RequestBody` and `RequestCookies`.
```
Comment (stephybun) [inline website/docs/r/web_application_firewall_policy.html.markdown:159]:
```suggestion
* `variable_name` - (Required) The name of the Match Variable. Possible values are `RemoteAddr`, `RequestMethod`, `QueryString`, `PostArgs`, `RequestUri`, `RequestHeaders`, `RequestBody` and `RequestCookies`.
```

**Review by stephybun — CHANGES_REQUESTED** [review]

**Review by stephybun — APPROVED** [review]
Thanks @wuxu92 LGTM 🌻

**Pull request merged**